src/context/locale/defaultLocale.ts

```diff
@@ -0,0 +1,6 @@
+import { AppLocale } from './appLocale.enum';
+
+const usersLocalce = navigator.language.substring(0, 2);
+
+export const defaultLocale: AppLocale =
+  usersLocalce === 'pl' || 'en' ? (usersLocalce as AppLocale) : ('en' as AppLocale);
```

Comment:
Trzy rzeczy:
1. `usersLocalce === 'pl' || 'en'` - ten warunek będzie zawsze prawdziwy, dlaczego?
2. Dobrze byłoby napisać tutaj jakieś bardziej generyczne rozwiązanie. Zauważ, że jeśli będziesz potrzebował dodać w przyszłości kolejny język, wtedy będziesz go musiał dodać i do enuma i do defaultLocale. Poszukaj w necie jak w TSie sprawdzić, czy dany typ jest wartością z enuma.
2. `'en' as AppLocale` - każde użycie `as` jest traktowaniem TypeScripta jako balast. A można to użyć inaczej, po prostu `AppLocale.en`.

Reply:
1. Jasne, bo to jest równoznaczne z (userLocale === 'pl') || true, czyli zawsze będzie true. Głupi błąd.
2. ok
3. ok

Reply:
Nie czaję dlaczego skoro defaultLocale jest typu AppLocale (obiekt en: 'en') to przyjmuje jako wartość AppLocale, który jest typu string?

Reply:
I w związku z tym nie wiem dlaczego nie działa mi to: `export const defaultLocale: AppLocale = usersLocalce in AppLocale ? usersLocalce  : AppLocale.en;`

Reply:
Tutaj trzeba załatwić po prostu: `export const defaultLocale: AppLocale = usersLocalce in AppLocale ? (usersLocalce as AppLocale) : AppLocale.en`

Reply:
Ok, do tego doszedłem, tylko myślałem jeszcze jak się pozbyć `(userLocale as AppLocale)`